src/widget/wbeatmenu.h

```diff
@@ -6,18 +6,24 @@
 #include "track/beat.h"
 #include "track/track.h"
 #include "util/parented_ptr.h"
+#include "widget/wcuemenupopup.h"
 
 class WBeatMenu : public QMenu {
     Q_OBJECT
   public:
     enum Option {
         SetDownbeat = 1 << 0,
+        CueMenu = 1 << 1
     };
     Q_DECLARE_FLAGS(Options, Option)
     WBeatMenu(UserSettingsPointer pConfig, QWidget* parent = nullptr);
 
     ~WBeatMenu() override = default;
 
+    // This function hides (not overrides) non-virtual QMenu::popup().
+    // This has been done on purpose to update menu before showing.
+    void popup(const QPoint& pos, QAction* at = nullptr);
```

Comment:
I think it would be better to update the QActions when the option flags are changed instead of having this confusing hiding of a standard QMenu method.